### PR TITLE
IBX-769: Reorder and hide for CKEditor toolbar groups and buttons, CustomStyleInline button support

### PR DESCRIPTION
**Pull request by damianz5**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-769](https://issues.ibexa.co/browse/IBX-769)
| **Improvement**| yes
| **New feature**    | yes
| **Target version** | master
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

 - Reorder and hide for CKEditor toolbar groups and buttons, 
 - show CustomStyleInline button only if inline custom styles exists,

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/lib/Configuration/Provider/CKEditor.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            }
+        }
+
+        array_pop($toolbars);
+
+        return array_values($toolbars);
+    }
+
+    /**
+     * @return string[] List of visible buttons
```
Comment (Steveb-p) [inline src/lib/Configuration/Provider/CKEditor.php:123]: Phpdoc does not match it's usage.

File: src/lib/Configuration/Provider/CKEditor.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            if (count($toolbarButtons)) {
+                $toolbars = array_merge(
+                    $toolbars,
+                    $toolbarButtons,
+                    [self::SEPARATOR],
+                );
+            }
+        }
+
+        array_pop($toolbars);
```
Comment (Steveb-p) [inline src/lib/Configuration/Provider/CKEditor.php:124]: I believe a comment explaining the purpose of this operation is in order here.

File: tests/lib/Configuration/Provider/CKEditorTest.php [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+    /**
+     * @covers \EzSystems\Tests\EzPlatformRichText\Configuration\Provider\CKEditorTest::createProvider
+     */
+    public function testGetConfiguration(): void
+    {
+        $provider = $this->createProvider();
+
+        $this->configResolver
+            ->expects($this->any())
```
Comment (Steveb-p) [inline tests/lib/Configuration/Provider/CKEditorTest.php:38]: This does not do anything.
```suggestion
```

File: tests/lib/Configuration/Provider/CKEditorTest.php [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+declare(strict_types=1);
+
+namespace EzSystems\Tests\EzPlatformRichText\Configuration\Provider;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use EzSystems\EzPlatformRichText\Configuration\Provider\CKEditor;
+use EzSystems\EzPlatformRichText\SPI\Configuration\Provider;
+use EzSystems\EzPlatformRichTextBundle\DependencyInjection\Configuration\Parser\FieldType\RichText;
+
+class CKEditorTest extends BaseProviderTestCase
```
Comment (Steveb-p) [inline tests/lib/Configuration/Provider/CKEditorTest.php:16]:
```suggestion
final class CKEditorTest extends BaseProviderTestCase
```

**Review by Steveb-p — CHANGES_REQUESTED** [review]
PR description declares that this does not introduce BC breaks. Can you confirm that users having previous configuration variant can safely upgrade without changing it?

What about existing editor extensions? Do they need to be updated?

I see that this goes into `master` branch, but if it introduces BC breaks in any way, then they should be documented and an upgrade path has to be provided.

**Comment by alongosz** [comment]
@damianz5 I feel like both JIRA issue and the title is to vague. Is this about extensibility of toolbar buttons or something more? Specifically what does it mean "Compatible with Custom plugins"?

**Comment by damianz5** [comment]
PR description updated

**Review by Steveb-p — APPROVED** [review]
Looks ok for me, although I'm not a fan of prefixing calls to global functions with `\`. Performance improvement is negligible (especially with OPCache).

**Review by ViniTou — APPROVED** [review]

File: tests/lib/Configuration/Provider/CKEditorTest.php [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+    /**
+     * @covers \EzSystems\Tests\EzPlatformRichText\Configuration\Provider\CKEditorTest::createProvider
+     */
+    public function testGetConfiguration(): void
+    {
+        $provider = $this->createProvider();
+
+        $this->configResolver
+            ->expects($this->exactly(1))
```
Comment (alongosz) [inline tests/lib/Configuration/Provider/CKEditorTest.php:38]: It's static, so should be called with `self::` and there's more specific method to achieve the same:
```suggestion
            ->expects(self::once())
```

**Review by alongosz — APPROVED** [review]

File: src/bundle/DependencyInjection/Configuration/Parser/FieldType/RichText.php [hunk 6]
```diff
@@ -193,11 +193,19 @@ public function addFieldTypeSemanticConfig(NodeBuilder $nodeBuilder)
 
         // RichText Toolbars configuration (defines list of Toolbars and Buttons enabled for current SiteAccess scope)
         $nodeBuilder
-            ->arrayNode(self::TOOLBARS_NODE_KEY)
-                ->useAttributeAsKey('name')
-                ->info('List of Toolbars and Buttons enabled for current SiteAccess scope.')
+            ->arrayNode(self::TOOLBAR_NODE_KEY)
```
Comment (adamwojs) [inline src/bundle/DependencyInjection/Configuration/Parser/FieldType/RichText.php:196]: This BC break needs to be mentioned in upgrade notes @ezsystems/documentation-team

**Review by adamwojs — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on IbexaDXP 4.0.0-dev.

**Pull request merged**